pyro/ops/ssm_gp.py

```diff
@@ -0,0 +1,113 @@
+            assert log_length_scale_init.shape == (num_gps,)
+        else:
+            log_length_scale_init = 0.01 * torch.randn(num_gps)
+
+        if log_kernel_scale_init is not None:
+            assert log_kernel_scale_init.shape == (num_gps,)
+        else:
+            log_kernel_scale_init = 0.01 * torch.randn(num_gps)
+
+        super(MaternKernel, self).__init__()
+
+        self.log_length_scale = nn.Parameter(log_length_scale_init)
+        self.log_kernel_scale = nn.Parameter(log_kernel_scale_init)
+
+        for x in range(self.state_dim):
+            for y in range(self.state_dim):
+                mask = torch.zeros(self.state_dim, self.state_dim)
+                mask[x, y] = 1.0
+                self.register_buffer("mask{}{}".format(x, y), mask)
+
+    def transition_matrix(self, dt):
+        """
+        Compute the (exponentiated) transition matrix of the GP latent space.
+        The resulting matrix has layout (num_gps, old_state, new_state), i.e. this
+        matrix multiplies states from the right.
+
+        See section 5 in reference [1] for details.
+
+        :param float dt: the time interval over which the GP latent space evolves.
+        :returns torch.Tensor: a 3-dimensional tensor of transition matrices of shape
+            (num_gps, state_dim, state_dim).
+        """
+        if self.nu == 1.5:
+            rho = self.log_length_scale.exp().unsqueeze(-1).unsqueeze(-1)
+            dt_rho = dt / rho
+            trans = (1.0 + root_three * dt_rho) * self.mask00 + \
+                    (-3.0 * dt_rho / rho) * self.mask01 + \
+                    dt * self.mask10 + \
+                    (1.0 - root_three * dt_rho) * self.mask11
+            return torch.exp(-root_three * dt_rho) * trans
+        else:
+            raise NotImplementedError
+
+    def stationary_covariance(self):
+        """
+        Compute the stationary state covariance. See Eqn. 3.26 in reference [2].
+
+        :returns torch.Tensor: a 3-dimensional tensor of covariance matrices of shape
+            (num_gps, state_dim, state_dim).
+        """
+        if self.nu == 1.5:
+            sigmasq = (2.0 * self.log_kernel_scale).exp().unsqueeze(-1).unsqueeze(-1)
+            rhosq = (2.0 * self.log_length_scale).exp().unsqueeze(-1).unsqueeze(-1)
+            p_infinity = sigmasq * self.mask00 + \
+                         (3.0 * sigmasq / rhosq) * self.mask11
+            return p_infinity
+
+    def process_covariance(self, A):
+        """
+        Given a transition matrix `A` computed with `transition_matrix` compute the
+        the process covariance as described in Eqn. 3.11 in reference [2].
+
+        :returns torch.Tensor: a batched covariance matrix of shape (num_gps, state_dim, state_dim)
+        """
+        assert A.shape == (self.num_gps, self.state_dim, self.state_dim)
+        p = self.stationary_covariance()
+        q = p - torch.matmul(A.transpose(-1, -2), torch.matmul(p, A))
```

Comment:
I believe you can abbreviate
```py
q = p - A.transpose(-1, -2) @ p @ A
```

Reply:
thanks but i really don't like using `@` 😬